src/pages/evm/nfts/NftViewer.vue

```diff
@@ -0,0 +1,211 @@
+}>();
+
+// data
+const nftTypes = {
+    image: 'image',
+    video: 'video',
+    audio: 'audio',
+    none: 'none',
+};
+
+// computed
+const nftType = computed(() => {
+    if (props.nft.imageSrcFull && !props.nft.audioSrc && !props.nft.videoSrc) {
+        return nftTypes.image;
+    } else if (props.nft.videoSrc) {
+        return nftTypes.video;
+    } else if (props.nft.audioSrc) {
+        return nftTypes.audio;
+    } else {
+        return nftTypes.none;
+    }
+});
+
+const showCoverImage = computed(() => nftType.value === nftTypes.image || props.previewMode);
+const showPlaceholderCoverImage = computed(() => !props.nft.imageSrcFull);
+
+const imageAlt = computed(() => {
+    const details = `${props.nft.name} ${props.nft.id}`;
+    if (nftType.value === nftTypes.image) {
+        return t('nft.img_alt', { nftInfo: details });
+    } else if (nftType.value === nftTypes.video && props.previewMode) {
+        return t('nft.img_alt_video_nft', { nftInfo: details });
+    }
+
+    return '';
+});
+
+const iconOverlayName = computed(() => {
+    const showIconOverlay = props.previewMode && nftType.value !== nftTypes.image;
+
+    if (!showIconOverlay) {
+        return '';
+    }
+
+    if (nftType.value === nftTypes.video) {
+        return 'o_play_arrow';
+    }
+
+    if (nftType.value === nftTypes.audio) {
+        return 'o_headphones';
+    }
+
+    return 'o_image_not_supported';
+});
+
+</script>
+
+<template>
+<div class="c-nft-viewer">
+    <div v-if="showCoverImage" class="c-nft-viewer__image-container">
+        <img
+            v-if="!showPlaceholderCoverImage"
+            :src="nft.imageSrcFull"
+            :alt="imageAlt"
+            class="c-nft-viewer__image"
+        >
+        <div v-else class="c-nft-viewer__placeholder-image"></div>
+
+        <template v-if="iconOverlayName">
+            <div class="c-nft-viewer__overlay-icon-bg shadow-2"></div>
+
+            <q-icon
+                :name="iconOverlayName"
+                size="lg"
+                color="primary"
+                class="c-nft-viewer__overlay-icon"
+            />
+        </template>
+    </div>
+
+    <div v-else-if="nftType === nftTypes.video" class="c-nft-viewer__video-container">
+        <!-- Implement video here https://github.com/telosnetwork/telos-wallet/issues/347 -->
+    </div>
+
+    <div v-else-if="nftType === nftTypes.audio" class="c-nft-viewer__audio-container">
+        <!-- Implement audio here https://github.com/telosnetwork/telos-wallet/issues/347 -->
+    </div>
+
+    <div v-else-if="nftType === nftTypes.none" class="c-nft-viewer__blank-container">
+
+    </div>
+</div>
+</template>
+
+<style lang="scss">
+.c-nft-viewer {
+    height: 100%;
+    width: 100%;
+    max-height: 270px;
+
+    &__image-container,
+    &__image,
+    &__video-container,
+    &__audio-container,
+    &__blank-container {
+        height: 100%;
+        width: 100%;
+    }
```

Comment:
I really love this way of styling things!
I learned it by reviewing your code and now is part of me :smiley: